components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -245,28 +246,33 @@ func newRTKMovementSensor(
 	if err := g.Start(ctx); err != nil {
 		return nil, err
 	}
+	g.mu.Lock()
+	defer g.mu.Unlock()
 	return g, g.lastError
 }
 
 func (g *RTKMovementSensor) setLastError(err error) {
-	g.errMu.Lock()
-	defer g.errMu.Unlock()
+	g.mu.Lock()
+	defer g.mu.Unlock()
 
 	g.lastError = err
 }
 
 // Start begins NTRIP receiver with specified protocol and begins reading/updating MovementSensor measurements.
 func (g *RTKMovementSensor) Start(ctx context.Context) error {
+	if err := g.nmeamovementsensor.Start(ctx); err != nil {
```

Comment:
[minor] could you add a TODO that requires testing out/cleaning up whether it is okay to have the gps `Start` before sending correction data via one of the comms protocols in the next two lines. 

I have a feeling doing them in parallel is the correct ultimate driver choice,  but something might be tied up in both functions right now.

Reply:
Ah, that hadn't occurred to me! I moved this one up so that we don't leak goroutines if this line fails, but it hadn't occurred to me that this line might give bad data if we don't have the corrections up and going first (which might still be a problem in the original ordering, too). TODO added!

Reply:
Note: for any TODOs added, please make it TODO(JIRA-XXX): where JIRA-XXX is some newly filed JIRA issue.

Reply:
Excellent feedback, thank you!